TRDataControl/Data/Remastered/BaseTRRDataCache.cs

```diff
@@ -40,30 +40,33 @@ public void SetData(TRDictionary<TKey, TRModel> pdpData, Dictionary<TKey, TAlias
 
     public void SetPDPData(TRDictionary<TKey, TRModel> pdpData, TKey sourceType, TKey destinationType)
     {
-        TKey translatedKey = TranslateKey(sourceType);
-        if (!_pdpCache.ContainsKey(sourceType))
+        lock (_pdpCache)
         {
-            string sourceLevel = GetSourceLevel(sourceType)
-                ?? throw new KeyNotFoundException($"Source PDP file for {sourceType} is undefined");
-
-            TRPDPControlBase<TKey> control = GetPDPControl();
-            TRDictionary<TKey, TRModel> models = control.Read(Path.Combine(PDPFolder, Path.GetFileNameWithoutExtension(sourceLevel) + _pdpExt));
-            if (models.ContainsKey(translatedKey))
-            {
-                _pdpCache[sourceType] = models[translatedKey];
-            }
-            else if (models.ContainsKey(destinationType))
-            {
-                _pdpCache[sourceType] = models[destinationType];
-            }
-            else
+            TKey translatedKey = TranslateKey(sourceType);
+            if (!_pdpCache.ContainsKey(sourceType))
```

Comment:
Are the calls to `ContainsKey` here with `sourceType` meant to actually be using `translatedKey`? Asking as it seems plausible and just because of where the call to `TranslateKey` happens.

Reply:
`sourceType` could be an alias, so we check for that in the cache - e.g. in `TR1RDataCache` we map secret types to actual TR1 types, so if caching these we need to store them by the alias otherwise we'd get a clash with the actual puzzle types.
It's not pretty.